File: DAY5/Parking.java
import java.util.Scanner;

public class Parking {
    int[] parkingSpots;
    int Numcar;

public Parking() {
parkingSpots = new int[20];
Numcar = 1;
        for (int index = 0; index < 20; index++) {
parkingSpots[index] = 0;
}
}

 public void showCarParking() {
  System.out.println("Parking lot:");
        for (int index = 0; index < 20; index++) {
  System.out.print("Spot " + (index + 1) + ": ");
   
if (parkingSpots[index] == 0) {
  System.out.println("No car");
} else {
    System.out.println("Car " + parkingSpots[index]);
}
}
}
    public void parkCar() {
        for (int index = 0; index < 20; index++) {
            if (parkingSpots[index] == 0) {
                parkingSpots[index] = Numcar;
       System.out.println("Car " + Numcar + " parked in spot " + (index + 1));
                Numcar++;
                return;
}
}

        System.out.println("Parking lot is full");
}

    public void takeOutCar(int spot) {
        if (spot < 1 || spot > 20) {
            System.out.println("Pick a number from 1 to 20");
            return;
}
        if (parkingSpots[spot - 1] == 0) {
            System.out.println("Spot " + spot + " is already empty");
            return;
}
        int car = parkingSpots[spot - 1];
        parkingSpots[spot - 1] = 0;
        System.out.println("Car " + car + " left spot " + spot + "!");
}

    private boolean isPositiveInteger(String input) {
        if (input == null || input.isEmpty()) {
            return false;
}
        for (int i = 0; i < input.length(); i++) {
            if (!Character.isDigit(input.charAt(i))) {
                return false;
}
}
        return true;
}

    public static void main(String[] args) {
        Parking parking = new Parking();
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("1. Park a car");
            System.out.println("2. Take a car out");
            System.out.println("3. Look at the parking lot");
            System.out.println("4. Stop playing");
            System.out.print("Pick a number: ");

String input = scanner.nextLine();
  if (!parking.isPositiveInteger(input)) {
       System.out.println("Please pick a number like 1, 2, 3, or 4!");
                continue;
}

 int choice = Integer.parseInt(input);
      if (choice < 1 || choice > 4) {
   System.out.println("Pick a number from 1 to 4!");
                continue;
}

if (choice == 1) {
   parking.parkCar();
        parking.showCarParking();
} else if (choice == 2) {
   System.out.print("Which spot (1 to 20)? ");
         String spotInput = scanner.nextLine();
if (!parking.isPositiveInteger(spotInput)) {
   System.out.println("Please pick a number like 1 or 2!");
 continue;
}
int spot = Integer.parseInt(spotInput);
    if (spot < 1 || spot > 20) {
        System.out.println("Pick a number from 1 to 20!");
continue;
}
parking.takeOutCar(spot);
     parking.showCarParking();
} else if (choice == 3) {
     parking.showCarParking();
} else if (choice == 4) {
System.out.println("Thanks for playing");
 break;
}
}
}
}

File: DAY5/Parking.py
class Parking:
    def __init__(self):
        self.parking_spots = [0] * 20
        self.num_car = 1

    def show_car_parking(self):
        print("Parking lot:")
        for index in range(20):
            print(f"Spot {index + 1}: ", end="")
            if self.parking_spots[index] == 0:
                print("No car")
            else:
                print(f"Car {self.parking_spots[index]}")

    def park_car(self):
        for index in range(20):
            if self.parking_spots[index] == 0:
                self.parking_spots[index] = self.num_car
                print(f"Car {self.num_car} parked in spot {index + 1}")
                self.num_car += 1
                return
        print("Parking lot is full")

    def take_out_car(self, spot):
        if spot < 1 or spot > 20:
            print("Pick a number from 1 to 20")
            return
        if self.parking_spots[spot - 1] == 0:
            print(f"Spot {spot} is already empty")
            return
        car = self.parking_spots[spot - 1]
        self.parking_spots[spot - 1] = 0
        print(f"Car {car} left spot {spot}!")

    def is_positive_integer(self, input_str):
        return input_str.isdigit() and int(input_str) > 0

    def get_parked_car_count(self):
        return sum(1 for spot in self.parking_spots if spot != 0)

def main():
    parking = Parking()
    while True:
        print("1. Park a car")
        print("2. Take a car out")
        print("3. Look at the parking lot")
        print("4. Stop playing")
        choice_input = input("Pick a number: ")

        if not parking.is_positive_integer(choice_input):
            print("Please pick a number from 1 to 4")
            continue

        choice = int(choice_input)
        if choice < 1 or choice > 4:
            print("Pick a number from 1 to 4!")
            continue

        if choice == 1:
            parking.park_car()
            parking.show_car_parking()
        elif choice == 2:
            spot_input = input("Which spot (1 to 20)? ")
            if not parking.is_positive_integer(spot_input):
                print("Please pick a number like 1 or 2!")
                continue
            spot = int(spot_input)
            if spot < 1 or spot > 20:
                print("Pick a number from 1 to 20!")
                continue
            parking.take_out_car(spot)
            parking.show_car_parking()
        elif choice == 3:
            parking.show_car_parking()
        elif choice == 4:
            print("Thanks for playing")
            break
        main()
